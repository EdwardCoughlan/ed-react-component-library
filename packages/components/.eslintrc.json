{
  "extends": [
    "plugin:@nrwl/nx/react",
    "../../.eslintrc.json",
    "airbnb",
    "airbnb-typescript"
  ],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": ["packages/components/tsconfig.*?.json"]
      },
      "rules": {
        "react/react-in-jsx-scope": "off",
        "react/require-default-props": "off",
        "import/prefer-default-export": "off"
      }
    },
    {
      "files": ["*.spec.ts", "*.spec.tsx", "*.stories.*"],
      "rules": {
        "import/no-extraneous-dependencies": "off"
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {
        "@typescript-eslint/explicit-function-return-type": ["error"]
      }
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
