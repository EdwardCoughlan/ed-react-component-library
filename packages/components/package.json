{
  "name": "@edwardcoughlan/components",
  "version": "0.0.1",
  "dependencies": {
    "@edwardcoughlan/tokens": "*"
  },
  "peerDependencies": {
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    "classnames": "^2.3.1"
  }
}
