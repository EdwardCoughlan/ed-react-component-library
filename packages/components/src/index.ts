export { Components } from './lib/components';
