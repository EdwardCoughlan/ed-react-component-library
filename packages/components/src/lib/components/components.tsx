import { FC } from 'react';
import { Tokens } from '@edwardcoughlan/tokens';
import cn from 'classnames';
import classes from './components.module.scss';

export const Components: FC = () => (
  <div className={cn(classes.molecules)}>
    <Tokens welcomeMessage="Hi from molecules" />
  </div>
);

export default Components;
