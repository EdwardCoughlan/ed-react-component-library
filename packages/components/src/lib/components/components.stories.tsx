import { Story, Meta } from '@storybook/react';
import { Components } from './components';

export default {
  component: Components,
  title: 'Components',
} as Meta;

const Template: Story = () => <Components />;

export const Primary = Template.bind({});
Primary.args = {};
