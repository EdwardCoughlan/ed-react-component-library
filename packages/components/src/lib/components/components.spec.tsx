import { render } from '@testing-library/react';

import { Components } from './components';

describe('Components', () => {
  it('should render successfully', () => {
    const { baseElement, container } = render(<Components />);
    expect(baseElement).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
