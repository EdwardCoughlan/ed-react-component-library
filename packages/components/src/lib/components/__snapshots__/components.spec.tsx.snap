// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components should render successfully 1`] = `
<div>
  <div
    class="molecules"
  >
    <div
      class="background"
    >
       
      <h1>
        Hi from molecules
      </h1>
    </div>
  </div>
</div>
`;
